src/pages/form-validation/phone-number-format/index.tsx

```diff
@@ -0,0 +1,236 @@
+import React, { useCallback, useState } from 'react';
+import {
+    AppBar,
+    FormControl,
+    Hidden,
+    IconButton,
+    InputLabel,
+    InputProps,
+    MenuItem,
+    Select,
+    TextField,
+    Toolbar,
+    Typography,
+} from '@material-ui/core';
+import { Menu } from '@material-ui/icons';
+import { makeStyles, Theme, useTheme } from '@material-ui/core/styles';
+import { useDispatch } from 'react-redux';
+import { TOGGLE_DRAWER } from '../../../redux/actions';
+
+type OnChangeHandler = InputProps['onChange'];
+
+type CountryDetails = {
+    code: string;
+    name: string;
+    placeholder: string;
+    maxLength: string;
+    errorCode: string;
+};
+
+const useStyles = makeStyles((theme: Theme) => ({
+    appbarRoot: {
+        padding: 0,
+    },
+    toolbarGutters: {
+        padding: '0 16px',
+    },
+    containerWrapper: {
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'start',
+        flex: '1 1 0',
+        backgroundColor: 'white',
+        height: 'calc(100vh - 64px)',
+    },
+    container: {
+        display: 'flex',
+        width: '100%',
+        maxWidth: 480,
+        margin: theme.spacing(4),
+        [theme.breakpoints.down('xs')]: {
+            margin: theme.spacing(2),
+            maxWidth: '100%',
+        },
+    },
+}));
+
+export const PhoneNumberFormatValidation = (): JSX.Element => {
+    const theme = useTheme();
+    const classes = useStyles(theme);
+    const dispatch = useDispatch();
+    const [phoneNumber, setPhoneNumber] = useState('');
+    const [countryCode, setCountryCode] = useState('US');
+    const [blurred, setBlurred] = useState(false);
+    const countries: CountryDetails[] = [
+        { code: 'US', name: '+1 (US)', placeholder: '### ### ####', maxLength: '12', errorCode: 'U.S.' },
+        { code: 'CA', name: '+1 (CA)', placeholder: '### ### ####', maxLength: '12', errorCode: 'Canadian' },
+        { code: 'RU', name: '+7 (RU)', placeholder: '### ### ## ##', maxLength: '13', errorCode: 'Russian' },
+        { code: 'EG', name: '+20 (EG)', placeholder: '# #######', maxLength: '9', errorCode: 'Egyptian' },
+        { code: 'IN', name: '+91 (IN)', placeholder: '#### ### ###', maxLength: '12', errorCode: 'Indian' },
+    ];
+
+    const transformUserInput = (value: string, country: string): string => {
```

Comment:
You can put this function outside `PhoneNumberFormatValidation` to prevent it from getting re-constructed every time this FC re-renders.

Or, have you considered using one of the functions suggested here on MUI's website? https://material-ui.com/components/text-fields/#integration-with-3rd-party-input-libraries

Reply:
That sounds good to me. You can leave a comment for our adopters to suggest them to look into those 3rd party tools